src/node_stat_watcher.cc

```diff
@@ -77,18 +77,13 @@ void StatWatcher::Initialize(Environment* env, Local<Object> target) {
 
 StatWatcher::StatWatcher(Environment* env, Local<Object> wrap)
     : AsyncWrap(env, wrap, AsyncWrap::PROVIDER_STATWATCHER),
-      watcher_(new uv_fs_poll_t) {
+      watcher_(nullptr) {
   MakeWeak();
-  uv_fs_poll_init(env->event_loop(), watcher_);
-  watcher_->data = static_cast<void*>(this);
 }
 
 
 StatWatcher::~StatWatcher() {
-  if (IsActive()) {
-    Stop();
-  }
-  env()->CloseHandle(watcher_, [](uv_fs_poll_t* handle) { delete handle; });
+  CHECK_EQ(watcher_, nullptr);
```

Comment:
Can you explain how we can be sure of this? Is this called somewhere besides from JS that I don’t see?

Also, as a side note: I think `StatWatcher` should just be a `HandleWrap`, if I’m not missing anything. That would solve this issue automatically.

Reply:
It has to have `Stop` called on it to reach the destructor in the first place, as far as I can tell. Is there an edge case I'm missing?

Reply:
Even without `unref`, yes, for Workers that’s the case:

```js
'use strict';
const { Worker, isMainThread, parentPort } = require('worker_threads');
const fs = require('fs');

if (isMainThread) {
  const worker = new Worker(__filename);
  worker.once('message', () => worker.terminate());
} else {
  const watcher = fs.watchFile(__filename, () => {});
  parentPort.postMessage('running');
}
```

I’ll PR that as a test file if that’s alright with you.

Reply:
Ah, that makes sense. Ok, let me update this to account for that. That said, it should still solve the GC issue.